[technical debt payment] use ink-picture, address vulnerabilities, and fix linter errors

endernoke opened the pull request:
Roadmap: #85
This PR will close #129 

## Changes
- Removed the in-house image components and use `ink-picture` instead, and removed a bunch of related dependencies
- Addressed some npm vulnerabilities
- Fixed some `xo` linter errors

## Todo
- Fix more linter errors

The number of vulnerabilities has gone down from 25 to 6. Most of these are from `instagram-typescript-api`, which I just realized hasn't been updated for a year 💀  We could only let them sit there and rot. The other solution is to fork the repo and maintain it but we probably shouldn't do it unless a maintain has a lot of free time lol

Fun fact: when you run `npx xo` on the current main branch, you will get exactly 1181 errors.  
With the changes now there's only 150 errors. The people shall rejoice and celebrate this marvelous achievement.  
Let's bring it down to 0 before this get merged 😃

supreme-gg-gg commented on instagram-ts/patches/ink+5.2.1.patch:
Is this file to be added intentionally? Why does the format look like diff result?

endernoke commented on instagram-ts/source/commands/index.tsx:
```diff
@@ -1,33 +1,19 @@
 	return (
 		<>
-			<Text color="magenta">{logo}</Text>
+			<BigText text="InstagramCLI" font="simple" colors={['#ff00ff']} />
```
done lol

endernoke commented:
> @endernoke I updated the `test.tsx` to stop using `App.tsx` which is just sitting rotting there doing nothing so it's removed (tech debt --;)

Nice, I added your changes while rebasing and tagged u as co author

> Regarding writing real tests in the future, I think our structure now is a bit difficult to test the commands directly since the mock can only be used to test the Views, not the command like `<Chat>` directly because it doesn't take in a client as props. I think testing the views would be fine because what matters is that the view renders and matches what we expect for the types (whcih are in the mock). What do you think about this?
> 
> We don't really need complex UI test cases, we just need to ensure people don't end up breaking prod 😭

Agreed, testing views with mock should be sufficient

Though I think the most important part of testing should be verifying interactivity, like making sure the thing doesn't break when you scroll up/down lmao, which is hard to test... Do you know of a way to test this?

supreme-gg-gg commented:
> > @endernoke I updated the `test.tsx` to stop using `App.tsx` which is just sitting rotting there doing nothing so it's removed (tech debt --;)
> 
> Nice, I added your changes while rebasing and tagged u as co author
> 
> > Regarding writing real tests in the future, I think our structure now is a bit difficult to test the commands directly since the mock can only be used to test the Views, not the command like `<Chat>` directly because it doesn't take in a client as props. I think testing the views would be fine because what matters is that the view renders and matches what we expect for the types (whcih are in the mock). What do you think about this?
> > We don't really need complex UI test cases, we just need to ensure people don't end up breaking prod 😭
> 
> Agreed, testing views with mock should be sufficient
> 
> Though I think the most important part of testing should be verifying interactivity, like making sure the thing doesn't break when you scroll up/down lmao, which is hard to test... Do you know of a way to test this?

Yea that's what I meant we couldn't really do "UI testing", I'm sure we can always just input a set of commands, but idk how to check the results, perhaps reference how frontend does UI testing I have no knowledge of it though

supreme-gg-gg commented:
> > Also I will create an issue to support Ink 6 on Pastel repo but it will 99.9% get ghosted, I suggest we just fork it and upgrade it to Ink 6 because Ink is very actively maintained with weekly releases, for long-term maintainability we shouldn't get stuck with Pastel which has not been maintained for half a year...
> 
> Sure go ahead let's farm more aura 🫡 I think pastel is now maintained by sindre sorhus, who is active, i just merged something with him a few days ago lol, so pastel might still have hope Perhaps if we don't get a reply within a week then let's fork it
> 
> I'll also open an issue on instagram private api to update their dependencies and fork it if i get ghosted

oh that's nice to hear, let's hope it goes well!
